Core/include/Acts/Seeding/InternalSpacePoint.hpp

```diff
@@ -71,17 +61,16 @@ class InternalSpacePoint {
 
 template <typename SpacePoint>
 inline InternalSpacePoint<SpacePoint>::InternalSpacePoint(
-    const SpacePoint& sp, const Acts::Vector3& globalPos,
+    std::size_t index, const SpacePoint& sp, const Acts::Vector3& globalPos,
```

Comment:
What do you mean? The space point indeed provide the global coordinate, but here are provided as inputs to the constructor since the space point API is not known. These info are stored internally in the `InternalSpacePoint` object and then retrieved by the seeding code (NEVER calling the external space points).

This is indeed a copy of all these variables ... thus the reason to get away from the `InternalSpacePoint` in the future